src/poetry/utils/flatdict.py

```diff
@@ -0,0 +1,32 @@
+from __future__ import annotations
+
+from typing import Any
+from collections.abc import Mapping
+
+
+def flatten_dict(obj: Mapping[str, Any], delimiter: str=".") -> Mapping[str, Any]:
```

Comment:
apart from the typing errors - per the pipeline - this doesn't belong in src, it's used in unit test only.

(for the same reason, end users never saw this dependency anyway - which isn't anything against this merge request exactly; but does say that if getting this clean is causing you much trouble, it may not be worth too much effort)

Reply:
Good point. Where would you suggest I put it. test/helpers ?

Reply:
Yes, tests/helpers is a much better choice.

Reply:
Now, it's duplicated.